src/Aspire.Hosting/Orchestrator/ApplicationOrchestrator.cs

```diff
@@ -148,9 +160,70 @@ static Task PublishUpdateAsync(ResourceNotificationService notificationService,
         }
     }
 
-    private async Task OnResourcesPrepared(OnResourcesPreparedContext _)
+    private async Task OnResourcesPrepared(OnResourcesPreparedContext context)
     {
-        await PublishResourcesWithInitialStateAsync().ConfigureAwait(false);
+        await PublishResourcesInitialStateAsync(context.CancellationToken).ConfigureAwait(false);
+    }
+
+    private async Task ProcessUrls(IResource resource, CancellationToken cancellationToken)
```

Comment:
I recently moved this into DcpExecutor. Is it duplicated here? If so, move to a place where it can be shared between the two types.

Also, can this now run twice for a resource? (here and in DcpExecutor). It looks like it would cause duplicate `ResourceUrlAnnotation` instances on a resource.

Reply:
Oh yeah, it's possible I didn't have that change locally. Let me sync and re-check.